### PR TITLE
Use @azure-tools/azcopy-node for download and upload

### PR DESCRIPTION
**Pull request by khanaffan**
The is a dedicated file handler that use azcopy for upload and download.

### CODE REVIEW

**Review by ramanujam-raman — APPROVED** [review]

File: core/backend-itwin-client/src/backend-itwin-client.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export * from "./imodelhub/LocalhostFileHandler";
 export * from "./imodelhub/AzureFileHandler";
 export * from "./UrlFileHandler";
 export * from "./StorageServiceFileHandler";
-
+export * from "./imodelhub/AzCopyFileHandler";
```
Comment (calebmshafer) [inline core/backend-itwin-client/src/backend-itwin-client.ts:13]: nit: can you move this to be alphabetically ordered?

File: core/backend-itwin-client/package.json [hunk 2]
```diff
@@ -94,7 +94,8 @@
     "js-base64": "^2.4.5",
     "jsonwebtoken": "^8.5.0",
     "openid-client": "^3.15.3",
-    "semver": "^5.5.0"
+    "semver": "^5.5.0",
+    "@azure-tools/azcopy-node" :"^1.0.0"
```
Comment (calebmshafer) [inline core/backend-itwin-client/package.json:98]: nit: can you move this up to the top of the list to alphabetize?

**Review by calebmshafer — CHANGES_REQUESTED** [review]
@khanaffan @ramanujam-raman not sure I see the reason to have the wrappers around the couple of the `fs` methods but they're all internal and already in AzureFileHandler as well.

File: core/backend-itwin-client/src/imodelhub/AzCopyFileHandler.ts [hunk 3]
```diff
@@ -0,0 +1,277 @@
+/*---------------------------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (calebmshafer) [reply core/backend-itwin-client/src/imodelhub/AzCopyFileHandler.ts:1]: What was the answer to this question?  Do we plan to deprecate the old one in favor of always using AzCopy or would we keep the old one for downloading via https?

[the remark this answers is not in the document]
Comment (khanaffan) [reply core/backend-itwin-client/src/imodelhub/AzCopyFileHandler.ts:1]: No i think we need to keep that around as fallback for sometime. Once we are comfortable with the new one we would switch to it completely.

File: core/backend-itwin-client/src/backend-itwin-client.ts [hunk 4]
```diff
@@ -10,7 +10,6 @@ export * from "./imodelhub/LocalhostFileHandler";
 export * from "./imodelhub/AzureFileHandler";
```
Comment (calebmshafer) [inline core/backend-itwin-client/src/backend-itwin-client.ts:10]: Looks like you remove the new file entirely, can you add it back here?

**Review by calebmshafer — CHANGES_REQUESTED** [review]

File: core/backend-itwin-client/src/backend-itwin-client.ts (hunk 4 above)
Comment (khanaffan) [reply core/backend-itwin-client/src/backend-itwin-client.ts:7]: done

**Review by calebmshafer — APPROVED** [review]

**Review by wgoehrig — APPROVED** [review]

**Pull request merged**